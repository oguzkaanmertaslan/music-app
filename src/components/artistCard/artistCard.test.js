import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import ArtistCard, {
  image,
  listener,
  artist,
  playcount,
  alt,
  onclick,
} from ".";

test("ArtistCard Text Controls", () => {
  render(<ArtistCard />);
  expect(screen.getByText(/artist/i)).toBeInTheDocument();
  expect(screen.getByText(/listeners :/i)).toBeInTheDocument();
  expect(screen.getByText(/playcount :/i)).toBeInTheDocument();
});

test("ArtistCard with props", () => {
  render(
    <ArtistCard
      image={image}
      listener={listener}
      playcount={playcount}
      alt={alt}
      artist={artist}
      onclick={onclick}
    />
  );
});
test("ArtistCard img controls", () => {
  render(<ArtistCard />);
  const image = screen.getByRole("img");
  expect(image).toBeInTheDocument();
  expect(image).toHaveClass("image-area");
});
test("ArtistCard Snapshot", () => {
  const tree = renderer
    .create(
      <ArtistCard
        image={image}
        listener={listener}
        playcount={playcount}
        alt={alt}
        artist={artist}
        onclick={onclick}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
