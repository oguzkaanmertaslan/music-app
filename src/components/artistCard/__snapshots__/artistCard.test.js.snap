// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistCard Snapshot 1`] = `
<div
  className="artist-card"
>
  <div
    className="left-area"
  >
    <div>
      <img
        className="image-area"
      />
    </div>
    <div
      className="left-text"
    >
      <span>
        Artist
      </span>
    </div>
  </div>
  <div
    className="right-area"
  >
    <div>
      listeners : 
    </div>
    <div>
      playcount : 
    </div>
  </div>
</div>
`;
